samples/drivers/watchdog/src/main.c

```diff
@@ -88,12 +93,19 @@ void main(void)
 	wdt_config.window.min = 0U;
 	wdt_config.window.max = WDT_MAX_WINDOW;
 
+#if (WDT_DISABLE_CALLBACK - 0)
+	printk("Callback in RESET_SOC disabled for this platform\n");
+	(void)wdt_callback;
```

Comment:
`wdt_config.callback` is left uninitialized in this path, so it can contain some garbage from the stack and thus a callback may get executed but from some unexpected location. This may get unnoticed because the watchdog reset comes right after, but...
Perhaps it will be the safest to add the default initialization of `wdt_config` to `{0}`, as for `CONFIG_WDT_MULTISTAGE` the structure contains also the `next` field which is not used in this sample.

Reply:
Absolutely correct; I didn't look closely enough at what this had been doing.

Switched to an initialization idiom that's more robust.  Also inverted the logic of the callback-support flag to be more clear.